src/test/java/com/amazon/dlic/auth/http/jwt/HTTPJwtAuthenticatorTest.java

```diff
@@ -275,8 +436,25 @@ public void testNonStringClaim() throws Exception {
         AuthCredentials creds = jwtAuth.extractCredentials(new FakeRestRequest(headers, new HashMap<String, String>()), null);
         Assert.assertNotNull(creds);
         Assert.assertEquals("Leonard McCoy", creds.getUsername());
-        Assert.assertEquals(1, creds.getBackendRoles().size());
-        Assert.assertTrue( creds.getBackendRoles().contains("123"));
+        Assert.assertEquals(0, creds.getBackendRoles().size());
+    }
+
+    @Test(expected = ElasticsearchSecurityException.class)
```

Comment:
Can you check if there is any other exception we can throw ? like IllegalArgumentException or something else 

ElasticsearchSecurityException doesn't sound good in this case

Reply:
I think `IllegalArgumentException("Expected type String or JSON array in the JWT for roles_key/roles_path")` is good.